### ISSUE TITLE
[2019 INTERSPEECH] End-to-End Neural Speaker Diarization with Permutation-Free Objectives

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#105]
가장 main이 되는 permutation-invariant training loss만 정리

## 1. Concept

먼저 해당 loss를 이해하는데 있어서 이해를 하는데 있어, `각 speech는 다 독립적이다`는 concept를 받아들여야 이해가 가능함

예를 들어 speaker A, B가 발화하는 1시간짜리 음성이 있다고 하면, VRAM에 load하기 위해 segments로 쪼개게 됨

3개의 20분 음성으로 쪼갰다고 하면, 쪼개진 음성에 대해서 A, B의 정보가 유기적으로 연결되는 것이 아니라 독립적인 화자로 생각해야함

즉 주어진 독립된 segment에 대해서 diarization을 잘하자...라는 것임

Diarization task의 목적인 who spoke when에서 who를 제대로 잡지 못하는 느낌이 있으나, 이는 고려하지 않음

## 2. Loss
<img width="500" alt="image" src="https://user-images.githubusercontent.com/49643709/204444873-6ea4251d-30ad-45be-ac7f-d30cc2ecefd8.png">
<img width="500" alt="image" src="https://user-images.githubusercontent.com/49643709/204445574-93e2b511-ad96-4145-9371-75978f5de074.png">

Task를 multi-label classification 문제로 봄

예를 들어 speaker A, B가 있는 상황에서 다음과 같이 정답 라벨이 주어짐
- Speaker가 발화하지 않는 경우 : [0, 0]
- Speaker A가 발화하는 경우 : [1, 0]
- Speaker B가 발화하는 경우 : [0, 1]
- Speaker A, B가 발화하는 경우 (speech overlap) : [1, 1]

Loss는 model의 output과 정답 label의 permutation 중에서 가장 작은 BCE loss가 loss가 됨

예를 들어 speaker A, B가 있는 상황에서 model의 output이 다음과 같음

> A : 0, 0.9, 0.8, 0.7, 0
> B : 0, 0, 0, 0.7, 0.9

정답 label은 다음과 같음

> A : 1, 1, 1, 0, 0
> B : 0, 0, 1, 1, 1

그렇다면 permutation을 통해 밑의 tensor도 정답이 됨

> A : 0, 0, 1, 1, 1
> B : 1, 1, 1, 0, 0

Model의 output과 permutation을 가한 정답 labels간의 BCE loss가 작은 loss를 선택하여 backprop하여 학습
